tests/test_aurora.py

```diff
@@ -0,0 +1,434 @@
+
+    return points, features
+
+
+@pytest.fixture
+def model_config():
+    """Basic model configuration for unstructured point data"""
+    return {
+        "input_features": 2,
+        "output_features": 2,
+        "embed_dim": 24,
+        "latent_dim": 128,
+        "max_points": 5000,
+        "max_seq_len": 1024,
+    }
+
+
+def test_swin3d_encoder(sample_3d_data, swin3d_config):
+    """Test Swin3D encoder with minimal 3D data"""
+    print("\n=== Testing Swin3D Encoder ===")
+    print(f"Input shape: {sample_3d_data.shape}")
+
+    encoder = Swin3DEncoder(**swin3d_config)
+    output = encoder(sample_3d_data)
+
+    print(f"Output shape: {output.shape}")
+    print(f"Output stats - Mean: {output.mean():.4f}, Std: {output.std():.4f}")
+
+    assert len(output.shape) == 3
+    assert output.shape[0] == sample_3d_data.shape[0]
+    assert output.shape[2] == swin3d_config["embed_dim"]
+    assert not torch.isnan(output).any()
+
+
+def test_decoder3d():
+    """Test 3D decoder with minimal dimensions"""
+    print("\n=== Testing 3D Decoder ===")
+    batch_size = 2
+    embed_dim = 32  # Reduced from 96 to 32
+    target_shape = (8, 8, 8)  # Reduced from 32 to 8
+
+    print(f"Target shape: {target_shape}")
+    decoder = Decoder3D(output_channels=1, embed_dim=embed_dim, target_shape=target_shape)
+    input_tensor = torch.randn(
+        batch_size, target_shape[0] * target_shape[1] * target_shape[2], embed_dim
+    )
+    print(f"Input tensor shape: {input_tensor.shape}")
+
+    output = decoder(input_tensor)
+    print(f"Output shape: {output.shape}")
+    print(f"Output stats - Mean: {output.mean():.4f}, Std: {output.std():.4f}")
+
+    assert output.shape == (batch_size, 1, *target_shape)
+    assert not torch.isnan(output).any()
+
+
+def test_perceiver_processor():
+    """Test Perceiver processor with minimal data"""
+    print("\n=== Testing Perceiver Processor ===")
+    batch_size = 2
+    seq_len = 16  # Reduced from 100 to 16
+    input_dim = 32  # Reduced from 96 to 32
+
+    print(
+        f"Input dimensions - Batch: {batch_size}, Sequence length: {seq_len}, Input dim: {input_dim}"
+    )
+    processor = PerceiverProcessor(latent_dim=64, input_dim=input_dim, max_seq_len=64)
+    input_tensor = torch.randn(batch_size, seq_len, input_dim)
+
+    output = processor(input_tensor)
+    print(f"Output shape: {output.shape}")
+    print(f"Output stats - Mean: {output.mean():.4f}, Std: {output.std():.4f}")
+
+    assert output.shape[0] == batch_size
+    assert output.shape[1] == 64
+    assert not torch.isnan(output).any()
+
+
+def test_full_pipeline_integration():
+    """Test minimal pipeline integration"""
+    print("\n=== Testing Full Pipeline Integration ===")
+    batch_size = 2
+    channels = 1
+    size = 8  # Reduced from 32 to 8
+    embed_dim = 32  # Reduced from 96 to 32
+
+    input_3d = torch.randn(batch_size, channels, size, size, size)
+    print(f"Input shape: {input_3d.shape}")
```

Comment:
In unit tests, we would prefer to not have print statements, if these are printing shapes to check they are correct, a better way to do it would be to have `asserts` to actually check it.